lib/self-service/self-service-manager.js

```diff
@@ -291,36 +305,47 @@ let _getIAMToken = function (iamToken, iamApiKey, iamTokenUrl) {
 	if (!iamApiKey) {
 		return Promise.reject("You must pass 'iamToken' to self-service-manager APIs or specify 'iamApiKey' in selfServiceManager init options.");
 	}
-	var reqOptions = {
-		url: iamTokenUrl,
-		method: "POST",
-		headers: {
-			"Content-Type": "application/x-www-form-urlencoded",
-			"Accept": "application/json"
-		},
-		form: {
-			"grant_type": "urn:ibm:params:oauth:grant-type:apikey",
-			"apikey": iamApiKey
-		}
-	};
+
 	return new Promise(function (resolve, reject) {
-		request(reqOptions, function (error, response, body) {
-			if (error) {
-				logger.error("Obtained IAM token failure: " + error.message);
-				reject(error.message);
-			} else {
+		(async () => {
+			try {
+				let formData = {
+					"grant_type": "urn:ibm:params:oauth:grant-type:apikey",
+					"apikey": iamApiKey
+				}
+
+				const reqParameters = {
+					headers: {
+						"Accept": "application/json",
+						"Content-Type": "application/x-www-form-urlencoded"
+					},
+					method: "POST",
+					body: jsonToURLencodedForm(formData),
+					responseType: 'json'
+				};
+
+				const response = await got(iamTokenUrl, reqParameters);
+				const body = parseJSON(response?.body);
+				
 				if (response.statusCode === 200) {
-					var IAMAccessToken = JSON.parse(body)["access_token"];
+					var IAMAccessToken = body["access_token"];
 					logger.debug("Obtained IAM token: " + IAMAccessToken);
 					resolve(IAMAccessToken);
-				} else {
+				}
+				else if (response.error) {
+					throw(response.error);
```

Comment:
I am throwing the `response.error` so it will go to the catch block which will then log the error reject the promise:
```
logger.error("Obtained IAM token failure: " + err.message);
reject(err.message);
```